smart_campus/app/templates/app/group_add_page.html

```diff
@@ -0,0 +1,33 @@
+{% extends 'app/base.html' %}
+{% load static %}
+
+{% block stylesheet %}
+  {{ block.super }}
+  <link rel="stylesheet" type="text/css" href="{% static "app/css/edit.css" %}">
+{% endblock %}
+
+{% block content %}
+  <div class="ui text container">
+    {% if messages %}
+      {% for message in messages %}
+      <div class="ui info message">
+        {{ message }}
+      </div>
+      {% endfor %}
+    {% endif %}
```

Comment:
Line 11 ~ line 17 appear over and over again.
Maybe you can take a look on how [flasky](https://github.com/miguelgrinberg/flasky/blob/master/app/templates/base.html#L56) implement it.

Reply:
I found the jinja2 has `marco` to solve this question.
For the django template language, it would be `include (with)`.

Reply:
https://stackoverflow.com/questions/31630319/macros-in-django-templates